ENIGMAsystem/SHELL/Universal_System/Resources/asset_index.cpp

```diff
@@ -0,0 +1,37 @@
+#include "asset_index.h"
+
+using enigma::assetMap;
+using enigma::assetTypes;
+
+namespace enigma_user {
+
+int asset_get_index(std::string resource, AssetType type) {
+ if (type == asset_any) {
+  for (const AssetType t : assetTypes) {
+    if (assetMap.count(t) == 0) continue;
+    const auto& m = assetMap.at(t);
+    auto it = m.find(resource);
+    if (it != m.end()) return it->second;
+    } 
+  } else {
+    if (assetMap.count(type) == 0) return -1;
+    const auto& m = assetMap.at(type);
+    auto it = m.find(resource);
+    if (it != m.end()) return it->second;
+  }
+  
+  return -1;
+}
+
+int asset_get_type(std::string resource) {
+  for (const AssetType t : assetTypes) {
+    if (assetMap.count(t) == 0) continue;
+    const auto& m = assetMap.at(t);
+    auto it = m.find(resource);
+    if (it != m.end()) return t;
```

Comment:
Could just use find both times here, instead of hashing `t` more than once. Some duplicate logic here too, which is ok for now.
```suggestion
    auto mit = assetMap.find(t);
    if (mit == assetMap.end()) continue;
    const auto& m = *mit;
    auto it = m.find(resource);
    if (it != m.end()) return t;
```